src/GraphQLParser/Extensions/ASTNodeExtensions.cs

```diff
@@ -0,0 +1,37 @@
+using GraphQLParser.AST;
+using GraphQLParser.Visitors;
+
+namespace GraphQLParser
+{
+    /// <summary>
+    /// Extension methods for <see cref="ASTNode"/>.
+    /// </summary>
+    public static class ASTNodeExtensions
+    {
+        /// <summary>
+        /// Counts the number of all nested nodes of the specified node along with that node itself.
+        /// </summary>
+        /// <param name="node">Parent node.</param>
+        /// <returns>The number of nodes.</returns>
+        public static int Count(this ASTNode node)
```

Comment:
We don't have any nodes with a Count property?  If so, this should be renamed so it does not conflict.  I feel also that this method name is not really descriptive of what it does.  The LINQ method called `Count()` simply counts direct descendants.  This counts all descendants.  We should have a more clear name here.  Perhaps `GetTotalCount()`.

Reply:
> We don't have any nodes with a Count property?

No such nodes. Changed to `AllNestedCount`. Not a fan of _Get_ prefixes for extension methods.